pkg/extender/extender_test.go

```diff
@@ -1534,3 +1546,476 @@ func extenderMetricsTest(t *testing.T) {
+
+	if err := driver.CreateCluster(6, nodes); err != nil {
+		t.Fatalf("Error creating cluster: %v", err)
+	}
+	pod := newPod("sharedV4ServiceAntihyperconvergence", map[string]bool{"sharedV4ServiceAntihyperconvergence": false})
+
+	provNodes := []int{0, 1, 2}
+	if err := driver.ProvisionVolume("sharedV4ServiceAntihyperconvergence", provNodes, 3, nil, true); err != nil {
+		t.Fatalf("Error provisioning volume: %v", err)
+	}
+
+	filterResponse, err := sendFilterRequest(pod, nodes)
+	if err != nil {
+		t.Fatalf("Error sending filter request: %v", err)
+	}
+	verifyFilterResponse(t, nodes, []int{0, 1, 2, 3, 4, 5}, filterResponse)
+
+	prioritizeResponse, err := sendPrioritizeRequest(pod, nodes)
+	if err != nil {
+		t.Fatalf("Error sending prioritize request: %v", err)
+	}
+	verifyPrioritizeResponse(
+		t,
+		nodes,
+		[]float64{defaultScore,
+			defaultScore,
+			defaultScore,
+			nodePriorityScore,
+			nodePriorityScore,
+			nodePriorityScore},
+		prioritizeResponse)
+}
+
+// Offline nodes should not be returned in filter response for pods using sharedv4 service volumes
+func sharedV4ServiceWithOfflineNodesAntihyperconvergenceTest(t *testing.T) {
+	nodes := &v1.NodeList{}
+	nodes.Items = append(nodes.Items, *newNode("node1", "node1", "192.168.0.1", "rack1", "a", "zone1"))
+	nodes.Items = append(nodes.Items, *newNode("node2", "node2", "192.168.0.2", "rack1", "a", "zone1"))
+	nodes.Items = append(nodes.Items, *newNode("node3", "node3", "192.168.0.3", "rack1", "a", "zone1"))
+	nodes.Items = append(nodes.Items, *newNode("node4", "node4", "192.168.0.4", "rack2", "b", "zone2"))
+	nodes.Items = append(nodes.Items, *newNode("node5", "node5", "192.168.0.5", "rack2", "b", "zone2"))
+	nodes.Items = append(nodes.Items, *newNode("node6", "node6", "192.168.0.6", "rack2", "b", "zone2"))
+
+	if err := driver.CreateCluster(6, nodes); err != nil {
+		t.Fatalf("Error creating cluster: %v", err)
+	}
+	pod := newPod("sharedV4ServiceWithOfflineNodesAntihyperconvergenceTest", map[string]bool{"sharedV4ServiceWithOfflineNodesAntihyperconvergenceTest": false})
+
+	provNodes := []int{0, 1, 2}
+	if err := driver.ProvisionVolume("sharedV4ServiceWithOfflineNodesAntihyperconvergenceTest", provNodes, 3, nil, true); err != nil {
+		t.Fatalf("Error provisioning volume: %v", err)
+	}
+
+	if err := driver.UpdateNodeStatus(0, volume.NodeOffline); err != nil {
+		t.Fatalf("Error setting node status to Offline: %v", err)
+	}
+
+	if err := driver.UpdateNodeStatus(3, volume.NodeOffline); err != nil {
+		t.Fatalf("Error setting node status to Offline: %v", err)
+	}
+
+	filterResponse, err := sendFilterRequest(pod, nodes)
+	if err != nil {
+		t.Fatalf("Error sending filter request: %v", err)
+	}
+	verifyFilterResponse(t, nodes, []int{1, 2, 4, 5}, filterResponse)
+}
+
+// Deploy both sharedv4 service volume and regular volumes
+// Verify nodes where pods using sharedv4 service volumes can be antihyperconverged takes higher score
+func multiVolumeAntihyperconvergenceTest(t *testing.T) {
```

Comment:
Can you add a multiVolume test where there are 6 nodes and the volume replicas are distributed in the following way:

Case 1:
- regularVolume: 0, 1, 2
- sharedv4SvcVolume: 2, 3, 4

priorityResponse: {2*nodePriority, 2*nodePriority, defaultScore, defaultScore, defaultScore}

Case 2: 

- regularVolume: 0, 1, 2
- sharedv4SvcVolume: 0, 1, 2

filterRequest should return {0, 1, 2}
prioritizeRequest should return {defaultScore, defaultScore, defaultScore}

Reply:
Case2: 
With preferRemoteOnlyNode enabled, filter will return (3,4, 5) in filter response
Without preferRemoteOnly, filter will return all nodes

Reply:
Added two more test cases. Handled case 2 with preferRemoteNodeOnly as I commented above